unity-ggjj/Assets/Scripts/BGScenes/BGSceneList.cs

```diff
@@ -1,21 +1,51 @@
 using System.Collections.Generic;
+using System.Linq;
 using UnityEngine;
 
 public class BGSceneList : MonoBehaviour
 {
-    private Dictionary<string, BGScene> _scenesInChildren;
-
+    [Tooltip("Drag a NarrativeScriptPlaylist here")]
+    [SerializeField] private NarrativeScriptPlaylist _narrativeScriptPlaylist;
+    
+    private readonly Dictionary<string, BGScene> _bgScenes = new Dictionary<string, BGScene>();
     private BGScene _activeScene;
 
     /// <summary>
-    /// Initializes the background scene dictionary
+    /// Gets all available BGScenes and instantiates them.
+    /// </summary>
+    public void Start()
+    {
+        foreach (var narrativeScript in _narrativeScriptPlaylist.NarrativeScripts)
+        {
+            InstantiateBgScenes(narrativeScript);
+        }
+
+        foreach (var narrativeScript in _narrativeScriptPlaylist.FailureScripts)
+        {
+            InstantiateBgScenes(narrativeScript);
+        }
+        
+        InstantiateBgScenes(_narrativeScriptPlaylist.GameOverScript);
+    }
+
+    /// <summary>
+    /// Instantiates BGScenes from a given narrative script and adds them to the dictionary
     /// </summary>
-    private void Awake()
+    /// <param name="narrativeScript">The narrative script to get BGScenes from</param>
+    private void InstantiateBgScenes(NarrativeScript narrativeScript)
     {
-        _scenesInChildren = new Dictionary<string, BGScene>();
-        foreach (BGScene scene in GetComponentsInChildren<BGScene>(true))
+        var backgroundScenes = narrativeScript.ObjectStorage.GetObjectsOfType<BGScene>();
+        foreach (var bgScene in backgroundScenes)
         {
-            _scenesInChildren.Add(new SceneAssetName(scene.gameObject.name), scene);
+            if (_bgScenes.Keys.Contains(bgScene.name))
```

Comment:
Nit: `Dictionary<K,V>`s contain a `.ContainsKey(K key)` method, which uses whatever underlying implementation is provided and should be quicker than iterating over the `Keys` and filtering them using LINQ.
This also should make `using System.Linq;` redundant.

```suggestion
if (_bgScenes.Keys.Contains(bgScene.name))
```

Reply:
Changed Keys.Contains to ContainsKey. I think thats what you meant? Your suggested change seem to just be removing white space.

Reply:
Oh whoops, yeah, sorry. 🙃 You did the right thing haha